### ISSUE TITLE
Random segfault when changing quality to source

### ISSUE THREAD

**Issue by dengelt** [vinszent/gnome-twitch#251]
I just had a segfault when I changed the quality to source (doesn't happen normally).

Backtrace:

```
#0  gt_twitch_stream_list_filter_quality (list=0x0, quality=<optimized out>)
    at ../src/gt-twitch.c:717
        ret = 0x0
        __func__ = "gt_twitch_stream_list_filter_quality"
#1  0x00005580bdc9a559 in streams_list_cb (source=<optimized out>, 
    res=<optimized out>, udata=<optimized out>) at ../src/gt-player.c:902
        __func__ = "streams_list_cb"
        stream_data = <optimized out>
        err = 0x5580bf13b780
#2  0x00007f6f046e0503 in ?? () from /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
No symbol table info available.
#3  0x00007f6f046e0539 in ?? () from /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
No symbol table info available.
#4  0x00007f6f041396aa in g_main_context_dispatch ()
   from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#5  0x00007f6f04139a60 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#6  0x00007f6f04139b0c in g_main_context_iteration ()
   from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#7  0x00007f6f046f572d in g_application_run ()
```

Seems like for some reason, the `gt_twitch_stream_list_filter_quality` function got `NULL` passed as the list argument.

EDIT:
Now that I think about it, that was when the stream went down and I used the quality options to "refresh" the stream, so the list was probably `NULL` because the stream was technically offline. I still think though that gnome-twitch should check the list pointer so it doesn't segfault.

**Comment by vinszent** [comment]
Usually Twitch returns a 404 when the stream is offline and the playlist is empty. But if there's anything I've learnt that is that Twitch's API is far from consistent 😔 .

Thanks for the stacktrace, I'll fix it this week and include it in the 0.4.1 release.

**Comment by Techwolf** [comment]
I got this error too. I think its due to trying to play a stream from the favorites list, but that stream is not up. It can be reproduced by waiting for a stream to end, go back and try to play the just ended stream. I noticed in consule output of <tr><td><b>type</b></td><td>error</td></tr>
<tr><td><b>error</b></td><td>Transcode does not exist</td></tr></table>'. I would paste more, but it has session and token data in it. I'll leave gdb running with this crash for while in case you need some more detail.
